### PR TITLE
Include proxy-only subnet with purpose=REGIONAL_MANAGED_PROXY when generating firewall rules for Ingress

### PR DESCRIPTION
**Pull request by gauravkghildiyal**
/assign @swetharepakula

### CODE REVIEW

File: pkg/loadbalancers/features/l7ilb.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func ILBSubnetSourceRange(cloud *gce.Cloud, region string) (string, error) {
 		if err != nil {
 			return "", fmt.Errorf("error comparing subnets: %v", err)
 		}
-		if subnet.Role == "ACTIVE" && subnet.Purpose == "INTERNAL_HTTPS_LOAD_BALANCER" && sameNetwork {
+		if subnet.Role == "ACTIVE" && (subnet.Purpose == "INTERNAL_HTTPS_LOAD_BALANCER" || subnet.Purpose == "REGIONAL_MANAGED_PROXY") && sameNetwork {
```
Comment (aojea) [inline pkg/loadbalancers/features/l7ilb.go:47]: is it possible that these two subnets exist at the same time and we have to return multiple `subnet.IpCidrRange`?

Comment (gauravkghildiyal) [reply pkg/loadbalancers/features/l7ilb.go:47]: That is a great question!

I don't think that's possible. Source https://cloud.google.com/load-balancing/docs/proxy-only-subnets: 

_"You can't have both an INTERNAL_HTTPS_LOAD_BALANCER and a REGIONAL_MANAGED_PROXY subnet in the same network and region, in the same way you can't have two REGIONAL_MANAGED_PROXY proxies or two INTERNAL_HTTPS_LOAD_BALANCER proxies._"

I anyways gave it a try and couldn't create two.

**Comment by aojea** [comment]
/lgtm

@swetharepakula for final approval

**Pull request merged**